src/ck3/data/prov_terrain.rs

```diff
@@ -0,0 +1,172 @@
+use std::hash::Hash;
+use std::path::PathBuf;
+
+use fnv::{FnvHashMap, FnvHashSet};
+
+use crate::block::Block;
+use crate::everything::Everything;
+use crate::fileset::{FileEntry, FileHandler};
+use crate::helpers::dup_error;
+use crate::item::Item;
+use crate::pdxfile::PdxFile;
+use crate::report::{warn, ErrorKey};
+use crate::token::Token;
+use crate::validator::Validator;
+use crate::Severity;
+
+use super::provinces::ProvId;
+
+#[derive(Clone, Debug, Default)]
+pub struct ProvinceTerrains {
+    provinces: FnvHashSet<ProvinceTerrain>,
```

Comment:
I'd like a comment here explaining that ProvinceTerrain is only hashed by its key. Otherwise this HashSet looks weird.

Reply:
No `ProvId` is used here as a hashmap key because there are also `default_*` ones at the top, which obviously cannot be parsed to ProvId`, but I do like to verify those too. I will add this to the comment.